invenio_app/limiter.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Invenio.
+# Copyright (C) 2019 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Flask Limiter functions."""
+
+from __future__ import absolute_import, print_function
+
+import pkg_resources
+from flask import current_app, request
+
+
+def limit_with_useragent_and_ip():
+    """Create key for the rate limiting."""
+    return str(request.user_agent) + request.remote_addr
+
+
+def set_rate_limit():
+    r"""Set rates for Flask limiter.
+
+    :return: a rate limit string with the Flask-Limiter format.
+
+    For more information regarding the format you can see `here
+    <https://flask-limiter.readthedocs
+    .io/en/stable/#rate-limit-string-notation>`_.
+
+    The order in which the rate will be evaluated is the following:
+
+    1)Initially the endpoint is going to be evaluated against the whitelisted
+    ones.If it has been marked as whitelisted then the custom limit for this
+    endpoint will be the one to be returned.
+
+    2)If the endpoint is not whitelisted and the flask-security package is
+    installed, it will evaluate if the user is logged in and if this is the
+    case it will also check if he is a privileged one comparing theire email
```

Comment:
Keep in mind that if you would think that the comment about renaming `RATELIMIT_PRIVILEGED_USERS` makes sense, this phrase has to change :)